aws-rds-dbinstance/src/main/java/software/amazon/rds/dbinstance/BaseHandlerStd.java

```diff
@@ -544,19 +580,48 @@ private void assertNoTerminalStatus(final DBInstance dbInstance) throws CfnNotSt
 
     protected boolean isDBInstanceStabilizedAfterMutate(
             final ProxyClient<RdsClient> rdsProxyClient,
-            final ResourceModel model
+            final ResourceModel model,
+            final CallbackContext context
     ) {
         final DBInstance dbInstance = fetchDBInstance(rdsProxyClient, model);
 
         assertNoTerminalStatus(dbInstance);
 
-        return isDBInstanceAvailable(dbInstance) &&
+        if (isDBInstanceAvailable(dbInstance) &&
                 isReplicationComplete(dbInstance) &&
                 isDBParameterGroupNotApplying(dbInstance) &&
                 isNoPendingChanges(dbInstance) &&
                 isVpcSecurityGroupsActive(dbInstance) &&
                 isDomainMembershipsJoined(dbInstance) &&
-                isPromotionTierUpdated(dbInstance, model);
+                isPromotionTierUpdated(dbInstance, model)
+        ) {
+            assertNoFailedEvents(rdsProxyClient, model, context);
+            return true;
+        }
+        return false;
+    }
+
+    protected void assertNoFailedEvents(
+            final ProxyClient<RdsClient> rdsProxyClient,
+            final ResourceModel model,
+            final CallbackContext context
+    ) {
+        final Instant fetchSince = context.getTimestamp(RESOURCE_UPDATED_AT);
+        if (fetchSince == null) {
+            // we did not timestamp it before,
```

Comment:
I think this has a chance of happening when sending Modify call to recover from storage-full scenario by invoking modify call with allocated-storage only. Should we then use named timestamps (i.e. `allocate-storage-start`)?